docs/contributing.rst

```diff
@@ -2,12 +2,635 @@
+    to fail pre-commit checks.
+
+3.4.1. Pre-commit setup
+^^^^^^^^^^^^^^^^^^^^^^^
+
+If you configured your Conda environment via the instructions in `3.3. Configure UXarray
+Conda Environment`_, you will have the :code:`pre-commit` package already installed in your
+environment (Otherwise, you will need to run :code:`conda install -c conda-forge pre_commit`
+to get it installed into your Conda environment for UXarray development). The only thing you
+will have to do additionally in order to set up pre-commit is to run the following command
+in your terminal in the UXarray root directory::
+
+    $ pre-commit install
+
+At this step, you are good to go with your pre-commit hooks to check your future commits.
+If, at any time, you'd like to run pre-commit hooks on your all files, you can run the
+following command::
+
+    $ pre-commit run --all-files
+
+3.5. Use Feature Branches
+-------------------------
+
+In your local clone, make a new branch off of the :code:`main` branch (This is way to go
+most of the time, but there might be specific cases where, for example, a branch is needed
+to be created off of another feature branch). Naming this branch, whenever applicable,
+like the following is not required but may be much helpful for tracking purposes:
+"issue_XXX" where XXX is the number of the issue or something that is representative of
+the changes you’re making, e.g. "do-this-work", "add-that-function", etc.
+
+Here are example commands that assume, you are checking out the :code:`main` branch first,
+pulling  from the remote server to have everything in your local up-to-date, and creating a new
+branch off of :code:`main`::
+
+    $ git checkout main
+    $ git pull
+    $ git checkout -b <new_branch>
+
+Once you create the new branch, you are good to go with your local changes in the UXarray
+directory!
+
+3.7. Local changes, commits, and pushes
+---------------------------------------
+
+The local development process can very basically be itemized as follows:
+
+1. Make change(s) to your local copy of the UXarray repository
+
+   .. note::
+        Please refer to `3.7.2. Common Elements of Most Pull Requests`_ to make sure your local
+        changes have all of the elements they should cover.
+
+2. Changes you’ve made locally must be “committed” (merged) to your local repository
+   (the .git subdirectory) using Git, but before the commit, you need to add them to the
+   "staged" changes for commit.
+
+   * You can see any uncommitted changes you’ve made to your local copy of the repository by
+     running the following command from anywhere (any directory) within the directory where
+     you ran :code:`git checkout`::
+
+        $ git status
+
+   * So, add changed file(s) into the staged changes to be included in a single commit::
+
+        $ git add PATH_TO_NEW_FILE
+
+     where `PATH_TO_NEW_FILE` is the path name of the newly created file.
```

Comment:
```suggestion
     where ``PATH/TO/NEW/FILE`` is the path name of the newly created file.
```
For this, it may also be better to use slashes instead to show the correct format for a path

Reply:
Both done